packages/grid/_modules_/grid/gridClasses.ts

```diff
@@ -1,62 +1,304 @@
+   */
+  scrollArea: string;
+  /**
+   * Styles applied to the left scroll area element.
+   */
+  scrollAreaLeft: string;
+  /**
+   * Styles applied to the right scroll area element.
+   */
+  scrollAreaRight: string;
+  /**
+   * Styles applied to the footer selected row count element.
+   */
+  selectedRowCount: string;
+  /**
+   * Styles applied to the sort icon element.
+   */
+  sortIcon: string;
+  /**
+   * Styles applied to the toolbar container element.
+   */
+  toolbarContainer: string;
+  /**
+   * Styles applied to the viewport element.
+   */
+  viewport: string;
+  /**
+   * Styles applied to the window element.
+   */
+  window: string;
+  /**
+   * Styles applied to the window container element.
+   */
+  windowContainer: string;
+  /**
+   * Styles applied to both the cell and the column header if `showColumnRightBorder={true}`.
+   */
+  withBorder: string;
+}
+
+export type GridClassKey = keyof GridClasses;
+
+export function getDataGridUtilityClass(slot: string): string {
+  return generateUtilityClass('MuiDataGrid', slot);
+}
 
 export const gridClasses = generateUtilityClasses('MuiDataGrid', [
-  'root',
-  'row',
-  'row--editing',
-  'cell',
-  'withBorder',
-  'cell--editing',
+  'autoHeight',
+  'booleanCell',
   'cell--editable',
-  'cell--withRenderer',
-  'cell--textLeft',
+  'cell--editing',
   'cell--textCenter',
+  'cell--textLeft',
   'cell--textRight',
-  'columnHeader',
-  'columnHeader--dragging',
-  'columnHeaderWrapper',
-  'scrollArea',
-  'scrollArea--left',
-  'scrollArea--right',
-  'columnsContainer',
-  'toolbarContainer',
-  'overlay',
-  'main',
-  'dataContainer',
-  'window',
-  'windowContainer',
-  'viewport',
-  'autoHeight',
-  'columnHeaderCheckbox',
+  'cell--withRenderer',
+  'cell',
   'cellCheckbox',
-  'columnHeader--sorted',
-  'columnHeader--sortable',
-  'columnHeader--moving',
-  'columnHeader--numeric',
-  'columnHeader--alignLeft',
+  'checkboxInput',
   'columnHeader--alignCenter',
+  'columnHeader--alignLeft',
   'columnHeader--alignRight',
+  'columnHeader--dragging',
+  'columnHeader--moving',
+  'columnHeader--numeric',
+  'columnHeader--sortable',
+  'columnHeader--sorted',
+  'columnHeader',
+  'columnHeaderCheckbox',
   'columnHeaderDraggableContainer',
+  'columnHeaderDropZone',
   'columnHeaderTitle',
-  'iconButtonContainer',
-  'sortIcon',
-  'filterIcon',
-  'menuIcon',
-  'menuIconButton',
   'columnHeaderTitleContainer',
-  'columnSeparator',
+  'columnHeaderWrapper',
+  'columnsContainer',
   'columnSeparator--resizable',
+  'columnSeparator--resizing',
+  'columnSeparator',
+  'dataContainer',
+  'editBooleanCell',
+  'editInputCell',
+  'filterIcon',
+  'footerContainer',
+  'iconButtonContainer',
   'iconSeparator',
+  'main',
+  'menuIcon',
+  'menuIconButton',
   'menuOpen',
-  'columnHeaderWrapper',
-  'editInputCell',
-  'editBooleanCell',
-  'booleanCell',
-  'checkboxInput',
+  'overlay',
+  'renderingZone',
+  'root',
```

Comment:
I wonder if we should keep the classes in a DOM order rather than an alphabetical one. For instance, *root* is the first one people will see in the DOM. In the core, we use the DOM order, but I would understand that it might not work well when they are many classes.

Reply:
DOM order would make more sense. I'll reorder them in #2465. For now, the alphabetical order is fine.